### PR TITLE
pythonPackages.uproot: init at 2.8.23.

### PR DESCRIPTION
**Pull request by ktf**
###### Motivation for this change

Package uproot is missing from the distribution.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1976,6 +1976,11 @@
     github = "kristoff3r";
     name = "Kristoffer Søholm";
   };
+  ktf = {
```
Comment (FRidh) [inline maintainers/maintainer-list.nix]: separate commit

File: pkgs/development/python-modules/uproot/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{lib, fetchFromGitHub, buildPythonPackage, numpy }:
+
+buildPythonPackage rec {
+  name = "uproot-${version}";
```
Comment (FRidh) [inline pkgs/development/python-modules/uproot/default.nix]: The `name` attribute is added by `buildPython*` when `pname` is added, and should therefore be removed.

File: pkgs/development/python-modules/uproot/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  #Checks are failing due to missing TTY, which won't exist.
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/scikit-hep/uproot";
+    description = "ROOT I/O in pure Python and Numpy.";
```
Comment (FRidh) [inline pkgs/development/python-modules/uproot/default.nix]: no dot at the end

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
@GrahamcOfBorg build python3.pkgs.uproot

**Comment by ktf** [comment]
I think I've addressed all the comments. Can you please check again?

**Comment by FRidh** [comment]
The expression needs to be called from `python-packages.nix`.

**Comment by ktf** [comment]
done. This is my first real contribution, sorry for being sloppy...

**Comment by FRidh** [comment]
@GrahamcOfBorg build python3.pkgs.uproot

File: pkgs/development/python-modules/uproot/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  pname = "uproot";
+  version = "2.8.16";
+  src = fetchFromGitHub {
+    owner  = "scikit-hep";
+    repo   = "uproot";
+    rev    = "${version}";
+    sha256 = "07f6h7zcvjc4ynqydp12cbq8c6bb8964s1i65cc25py4mg46f4my";
+  };
+
+  # Why does it propagate packages that are used for testing?
```
Comment (dotlambda) [inline pkgs/development/python-modules/uproot/default.nix]: numpy is listed in install_requires, you're sure it isn't a runtime dependency?

File: pkgs/development/python-modules/uproot/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{lib, fetchFromGitHub, buildPythonPackage, numpy }:
+
+buildPythonPackage rec {
+  pname = "uproot";
+  version = "2.8.16";
+  src = fetchFromGitHub {
```
Comment (dotlambda) [inline pkgs/development/python-modules/uproot/default.nix]: Please use `fetchPypi` instead: https://pypi.org/project/uproot/. This allows easier automatic updating.

File: pkgs/development/python-modules/uproot/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  # Why does it propagate packages that are used for testing?
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  #Checks are failing due to missing TTY, which won't exist.
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/scikit-hep/uproot";
```
Comment (dotlambda) [inline pkgs/development/python-modules/uproot/default.nix]: no quotes

**Review by dotlambda — CHANGES_REQUESTED** [review]
Did you intentionally not include the dependencies listed [here](https://github.com/scikit-hep/uproot#recommended-dependencies)?

File: pkgs/development/python-modules/uproot/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    rev    = "${version}";
+    sha256 = "07f6h7zcvjc4ynqydp12cbq8c6bb8964s1i65cc25py4mg46f4my";
+  };
+
+  # Why does it propagate packages that are used for testing?
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  #Checks are failing due to missing TTY, which won't exist.
```
Comment (dotlambda) [inline pkgs/development/python-modules/uproot/default.nix]: The tests run just fine for me. Can you please check again?

File: pkgs/development/python-modules/uproot/default.nix (hunk 4 above)
Comment (jpivarski) [reply pkgs/development/python-modules/uproot/default.nix]: You can't do anything sensible in uproot without Numpy. It's an absolute requirement. I hope the setup.py reflects that.

Many of the other libraries that uproot might use are optional, particularly compression codecs that aren't found in all ROOT files.

Comment (dotlambda) [reply pkgs/development/python-modules/uproot/default.nix]: Yes, setup.py reflects this. But the comment

> Why does it propagate packages that are used for testing?

doesn't make sense in that case.

**Comment by ktf** [comment]
I've done the requested changes.

@dotlambda I've tried adding `lz4` and `lama`, however they seem not to be picked up when building. I am using an overlay with the following to test:

```
uproot = import ../nixpkgs/pkgs/development/python-modules/uproot {
     inherit (self) lib;
     inherit (super.python27.pkgs) fetchPypi;
     inherit (super.python27.pkgs) buildPythonPackage numpy lz4;
     inherit (super) lzma;
};
```

and I of course added `lz4` and `lzma` to the `propagatedBuildInputs`. Am I doing something trivially wrong?

**Comment by jpivarski** [comment]
If it's Python 3, there might not be an lzma package because that's in the standard library now.

**Comment by ktf** [comment]
@jpivarski in my test it's `python2.7`. I suspect it's something nix related, actually.

File: pkgs/development/python-modules/uproot/default.nix [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{lib, fetchPypi, buildPythonPackage, numpy }:
+
+buildPythonPackage rec {
+  pname = "uproot";
+  version = "2.8.18";
+  name = "${pname}-${version}";
```
Comment (dotlambda) [inline pkgs/development/python-modules/uproot/default.nix]: drop this line

**Comment by dotlambda** [comment]
By looking at the code, e.g. https://github.com/scikit-hep/uproot/blob/47b981e6ce0b5913a763d379eae1c0337a898a01/tests/test_compression.py#L34-L41, it should just work when adding the lz4 (and lzma in the case of python 2) *python* packages to `propagatedBuildInputs`.

File: pkgs/development/python-modules/uproot/default.nix (hunk 8 above)
Comment (ktf) [reply pkgs/development/python-modules/uproot/default.nix]: you mean the `name` one? If I do I get an error. Am I using a too old version of `buildPythonPackage`?

Comment (dotlambda) [reply pkgs/development/python-modules/uproot/default.nix]: `buildPython*` has been setting `name` automatically for quite some time. Which error are you getting?

File: pkgs/development/python-modules/uproot/default.nix [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  };
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  meta = with lib; {
+    homepage = https://github.com/scikit-hep/uproot;
+    description = "ROOT I/O in pure Python and Numpy";
+    license = with licenses; [ bsd3 ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/uproot/default.nix]: just `licenses.bsd3`

**Comment by dotlambda** [comment]
@ktf Now what's left to do is naming your second commit (and your PR) `pythonPackages.uproot: init at <version>`.

@GrahamcOfBorg build python2.pkgs.uproot python3.pkgs.uproot

**Comment by ktf** [comment]
@dotlambda done.

**Pull request merged**